Remove binary tarball references on download page

absurdfarce opened the pull request:
Removing links to the current binary tarball and replacing them with links to the source tarball.  References to Maven Central will be used for those wanting the binary artifacts.

absurdfarce committed:
Update driver download section

patch by Bret McGuire; reviewed by mck and Andy Tolbert
reference: https://github.com/apache/cassandra-website/pull/304

absurdfarce commented on site-content/source/modules/ROOT/pages/download.adoc:
```diff
@@ -92,10 +92,9 @@ https://www.apache.org/dyn/closer.lua/cassandra/4.0.19/apache-cassandra-4.0.19-b
-(https://www.apache.org/dyn/closer.lua/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source.tar.gz[source,window=blank]: https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source.tar.gz.asc[pgp,window=blank], https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-sour ...
+(https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source.tar.gz.asc[pgp,window=blank], https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source.tar.gz.sha256[sha256,window=blank], https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source ...
 
 Java drivers for Cassandra are also available on
```
I debated whether this verbiage should be changed to something like "Binary artifacts for Java driver for Cassandra are available at Maven Central..."  Presumably if we do that we'd want to update the verbiage for the link above to clearly indicate that it's a _source_ release.

The golang links below _don't_ do this, but then again Go doesn't really have a binary v. source release distinction to worry about... so it's not the greatest example here.

absurdfarce commented:
Result of a local build with the changes to this point in the PR are:

<img width="802" height="366" alt="cassandra_website" src="https://github.com/user-attachments/assets/f1434c2e-763e-4bf6-a5f6-ce916c38ad8a" />

michaelsembwever commented on site-content/source/modules/ROOT/pages/download.adoc:
```diff
@@ -92,10 +92,9 @@ https://www.apache.org/dyn/closer.lua/cassandra/4.0.19/apache-cassandra-4.0.19-b
 
-(https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1.tar.gz.asc[pgp,window=blank], https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1.tar.gz.sha256[sha256,window=blank], https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1.tar.gz.sha512[sha512 ...
-(https://www.apache.org/dyn/closer.lua/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source.tar.gz[source,window=blank]: https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source.tar.gz.asc[pgp,window=blank], https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-sour ...
+(https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source.tar.gz.asc[pgp,window=blank], https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source.tar.gz.sha256[sha256,window=blank], https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source ...
```
```suggestion
(https://www.apache.org/dyn/closer.lua/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source.tar.gz[source,window=blank]: https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source.tar.gz.asc[pgp,window=blank], https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source.tar.gz.sha256[sha256,window=blank], https://downloads.apache.org/cassandra/cassandra-java-driver/4.19.1/apache-cassandra-java-driver-4.19.1-source.tar.gz.sha512[sha512,window=blank])
```

Opinions:

I would keep the `(source: pgp, sha256, sha512)` format of the line, keeping it consistent to the other download items on the page.

Optionally going further in removing the link of the green button (since that button on the rest of the page always downloads a binary artefact.

absurdfarce replied:
The format that was used here is identical to what's used with the Go driver just below it; I literally copied that block and updated the URLs.  I'm happy to update the links to add a leading "source:" before the links to the various checksums but it's also worth pointing out that it isn't the case that everything else on this page downloads a binary artifact.  That's what I was getting at when I mentioned that my proposal wound up making the Java driver look a lot like the Go driver in terms of how it's represented on this download page.

In unrelated news: if we're going to change to "www.apache.org/dyn/" for the source tarball do we not want to use the same structure for the checksum files?

In that same vein: if we change to "www.apache.org/dyn/" am I correct in saying we'll want to do the same for the Go driver?

tolbertam reviewed (approved):
👍 as is, and also 👍 with @michaelsembwever's suggested change as well.  Thank you @absurdfarce!

absurdfarce commented:
So after applying the suggested changes from @michaelsembwever I now have:

<img width="678" height="312" alt="download-update1" src="https://github.com/user-attachments/assets/ed181f98-f97a-4232-b0d4-42fda966cb34" />

If we can get clarity on the following points I think we're good here:

* Do we want to apply the "www.apache.org/dyn/" changes to the checksum files as well?
* Do we want to update the Go download links to use "www.apache.org/dyn/"?
* Should we mark the Go download link to be marked as "source" since it's also a source tarball?
* Maybe I misunderstood, but it sounded like there was an argument that we shouldn't keep the version number box (the one just above the "source" field) if we don't have a binary download.  Is this understanding correct?
  * If so presumably the same rationale applies to the Go driver?

michaelsembwever commented:
> Do we want to apply the "www.apache.org/dyn/" changes to the checksum files as well? 

No.   Checksums, keys and signatures must always be via a https://downloads.apache.org url
ref: see asf release policy.

The release artefact itself should use the https://dlcdn.apache.org/  url


> Should we mark the Go download link to be marked as "source" since it's also a source tarball?

Yes.  Good catch.
(stretch goal:  make the Go and Java download tiles on the same line.)


> Maybe I misunderstood, but it sounded like there was an argument that we shouldn't keep the version number box (the one just above the "source" field) if we don't have a binary download. Is this understanding correct?

Keep the box, just remove its link (as that's not consistent UX with the other download tiles on the page).  But this is opinionated and really doesn't matter either way.

> If so presumably the same rationale applies to the Go driver?

Yes.

absurdfarce commented:
Okay, fixes suggested above applied (along with a few extras).  Here's what I have now:

<img width="1270" height="664" alt="download-update2" src="https://github.com/user-attachments/assets/c4693e72-ce17-4905-b264-06c80efbe5cc" />

From less controversial to moreso:

* Added "source" link to Go driver so that it matches up what we're doing with the Java driver
* Both drivers now go name/description/last release data... gocql didn't have a last release date and Java had them in the other order.  The usage here is more consistent with other entities on the page.
* Moved both drivers onto a single row
  * With adequate buffering between this row and the "Unmaintained older versions..." row just above it
  * Generalizes nicely once Python and C/C++ join the fold
* I kept the button + link for both drivers
  * In each case they're still a link to the source file... which does differ from the other buttons
  * A button with no link here just seemed strange to me
  * Tried to mitigate any weirdness by explicitly changing the text to indicate that it's a source download

Thoughts?

absurdfarce commented on site-content/source/modules/ROOT/pages/download.adoc:
```diff
@@ -77,45 +77,45 @@ https://www.apache.org/dyn/closer.lua/cassandra/4.0.19/apache-cassandra-4.0.19-b
+https://www.apache.org/dyn/closer.lua/cassandra/cassandra-gocql-driver/2.0.0/apache-cassandra-gocql-driver-2.0.0.tar.gz[2.0.0 (source),window=blank]
 
-(https://downloads.apache.org/cassandra/cassandra-gocql-driver/2.0.0/apache-cassandra-gocql-driver-2.0.0.tar.gz.asc[pgp,window=blank], https://downloads.apache.org/cassandra/cassandra-gocql-driver/2.0.0/apache-cassandra-gocql-driver-2.0.0.tar.gz.sha256[sha256,window=blank], https://downloads.apache.org/cassandra/cassandra-gocql-driver/2.0.0/apache-cassandra-gocql-driver-2.0.0.tar.gz.sha512[sha512 ...
+(https://www.apache.org/dyn/closer.lua/cassandra/cassandra-gocql-driver/2.0.0/apache-cassandra-gocql-driver-2.0.0.tar.gz[source,window=blank]: https://downloads.apache.org/cassandra/cassandra-gocql-driver/2.0.0/apache-cassandra-gocql-driver-2.0.0.tar.gz.asc[pgp,window=blank], https://downloads.apache.org/cassandra/cassandra-gocql-driver/2.0.0/apache-cassandra-gocql-driver-2.0.0.tar.gz.sha256[sha2 ...
```
I'm wondering if this shouldn't be "source, pgp, sha256, sha512" rather than "source: pgp, sha256, sha512" (both here and on the Java driver above).  The single semicolon there seems kinda jarring... feels like we're calling out the source release for no clear reason.

michaelsembwever replied:
"source: pgp, sha256, sha512" makes more sense to me. semicolon indicates primary to auxiliary

michaelsembwever replied:
but given the button above it now clearly says "(SOURCE)"  then i would just simplify this to "pgp, sha256, sha512"

absurdfarce replied:
Oooooohhhh, I like that _much_ better @michaelsembwever!  Screenshots to follow but unless there's something super-strange there I'm going with that.

absurdfarce commented:
While I was here I updated the Java driver version to the just-released 4.19.2.

@tolbertam and/or @michaelsembwever ... anything else you guys want to see here or am I okay to merge this?

michaelsembwever commented:
all good from me @absurdfarce

absurdfarce commented:
Screenshot based on latest text as suggested by @michaelsembwever above:

<img width="1272" height="342" alt="download-update3" src="https://github.com/user-attachments/assets/d6af6c33-b5d9-4cc6-9128-4e65c1643f99" />

That's the best I've seen so far... I'm going with it.

absurdfarce commented:
I have no idea what happened but my last commit implementing the change suggested by @michaelsembwever just disappeared.  Opening a new PR to fix that.
